tests/vmipreset_test.go

```diff
@@ -367,9 +373,61 @@ var _ = Describe("[rfe_id:609][crit:medium][vendor:cnv-qe@redhat.com][level:comp
 			}
 		})
 	})
+
+	Context("Preset Lifecycle", func() {
+		var preset *v1.VirtualMachineInstancePreset
+		presetNamePrefix := "vmi-preset-small-"
+		selectorKey := "kubevirt.io/vmPreset"
+		selectorLabel := "vmi-preset-small"
+
+		BeforeEach(func() {
+			selector := k8smetav1.LabelSelector{MatchLabels: map[string]string{selectorKey: selectorLabel}}
+			memory, _ := resource.ParseQuantity("64M")
+			preset = &v1.VirtualMachineInstancePreset{
+				ObjectMeta: k8smetav1.ObjectMeta{GenerateName: presetNamePrefix},
+				Spec: v1.VirtualMachineInstancePresetSpec{
+					Selector: selector,
+					Domain: &v1.DomainSpec{
+						Resources: v1.ResourceRequirements{Requests: k8sv1.ResourceList{
+							"memory": memory}},
+					},
+				},
+			}
+		})
+
+		It("[test_id:617][rfe_id:609] should create and delete preset", func() {
+			By("Creating preset")
+			err := virtClient.RestClient().Post().Resource("virtualmachineinstancepresets").Namespace(tests.NamespaceTestDefault).Body(preset).Do().Error()
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Checking that preset was created")
+			newPreset, err := getPreset(virtClient, presetNamePrefix)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Deleting preset")
+			err = virtClient.RestClient().Delete().Resource("virtualmachineinstancepresets").Namespace(tests.NamespaceTestDefault).Name(newPreset.GetName()).Do().Error()
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Checking preset was deleted")
+			waitForPresetDeletion(virtClient, presetNamePrefix)
+		})
+	})
+
 })
 
-func waitForPreset(virtClient kubecli.KubevirtClient, prefix string) v1.VirtualMachineInstancePreset {
+func getPreset(virtClient kubecli.KubevirtClient, prefix string) (*v1.VirtualMachineInstancePreset, error) {
+	presetList := v1.VirtualMachineInstancePresetList{}
+	err := virtClient.RestClient().Get().Resource("virtualmachineinstancepresets").Namespace(tests.NamespaceTestDefault).Do().Into(&presetList)
```

Comment:
there's no reason to do a list if you know the preset's name.

```
preset, err := virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Get(presetName, metav1.GetOptions{})
```

Reply:
I agree, except that we *don't* know the name in all cases. We're using GenerateName in at least some subset of the tests.

Reply:
ah, alrighty then.